keps/sig-api-machinery/20190426-crd-defaulting.md

```diff
@@ -121,6 +123,23 @@ The `default` field in the CRD types is considered alpha quality. We will add a
 
 [Kubebuilder's crd-gen](https://github.com/kubernetes-sigs/controller-tools/tree/master/cmd/crd) can make use of this feature by adding another tag, e.g. `// +default=<arbitrary-json-value>`. Defaults are arbitrary JSON values, which must also validate (types are checked during CRD creation and update, value validation is checked for requests, but not for etcd reads) and are not subject to pruning (defaulting happens after pruning).
 
+### Validation
+
+Default values must be pruned, i.e. must not have fields which are not specified by the given structural schema, with the exception of
+ 
+* defaults inside of `.metadata` of an embedded resource (`x-kubernetes-embedded-resource: true`)
+* defaults which span `.metadata` of an embedded resource.
```

Comment:
what about defaults that span `x-preserve-unknown-fields`?

Reply:
Implied by "must be pruned", which depends on  `x-preserve-unknown-fields`.

Reply:
so pruning would leave those alone, and they would validate because there was no delta between the pruned and unpruned values? probably worth stating explicitly and adding a testcase to cover it